In [3]:
def f(x):
    return (x + 1) / (x - 1)

values = [0, 0.5, 0.9, 1.1, 1.5, 2]

for x in values:
    print(x, f(x))

0 -1.0
0.5 -3.0
0.9 -19.000000000000004
1.1 20.999999999999982
1.5 5.0
2 3.0


why x = 1 is not included in values ?
- because when x=1 the f(x) will become 2/0 which make the f(x)=DNE

In [4]:
def f(x):
    return (x + 1) / (x - 1)

values = [0, 0.5, 0.9, 0.99, 1.01, 1.1, 1.5, 2]

for x in values:
    print(x, f(x))

0 -1.0
0.5 -3.0
0.9 -19.000000000000004
0.99 -198.99999999999983
1.01 200.9999999999998
1.1 20.999999999999982
1.5 5.0
2 3.0


for 0.99 the f(0.99)= -199.
For f(1.01)= 201
So approaching 1 from the left f(x) become large negative. And when it approach from the right, it become a large positive number



Exercise 2

In [5]:
def placement(score):
    if score <= 1:
        return "Algebra"
    elif score <= 3:
        return "Trigonometry"
    else:
        return "Calculus"


print(placement(1))
print(placement(3))
print(placement(5))


students = [
    ("Maya", 20681001, 5),
    ("Noah", 20681002, 1),
    ("Olivia", 20681003, 3),
    ("Liam", 20681004, 4),
    ("Sophia", 20681005, 2),
    ("Ethan", 20681006, 0)
]

algebra = ["Algebra", []]
trig = ["Trigonometry", []]
calculus = ["Calculus", []]

math_courses = (algebra, trig, calculus)

for student in students:
    name, student_id, score = student

    course = placement(score)

    if course == "Algebra":
        algebra[1].append((name, student_id))
    elif course == "Trigonometry":
        trig[1].append((name, student_id))
    else:
        calculus[1].append((name, student_id))

print(math_courses)

Algebra
Trigonometry
Calculus
(['Algebra', [('Noah', 20681002), ('Ethan', 20681006)]], ['Trigonometry', [('Olivia', 20681003), ('Sophia', 20681005)]], ['Calculus', [('Maya', 20681001), ('Liam', 20681004)]])


Ex3

In [6]:
i = 0
total = 0

while i < len(math_courses):
    total = total + len(math_courses[i][1])
    i = i + 1

print(total)

6


The program should be 6, cause ex 2 there are 6 students. 2 in Algebra, 2 in Trig, 2 in Cal. The loop will count students in each class and add them